File: mod/library/server/handle-message/exporter/process-mod/process_mod_icon.lua
local output_image = SMODS.load_file("mod/server/handle-message/exporter/_common/output_image.lua")()

local function get_mod_icon_info(mod)
	local tag_atlas = mod.prefix and mod.prefix .. '_modicon' or 'modicon'
	local tag_pos = { x = 0, y = 0 }
	if not mod.can_load and mod.load_issues and mod.load_issues.prefix_conflict then
		tag_atlas = 'mod_tags'
		tag_pos = { x = 0, y = 0 }
	end
	return tag_atlas, tag_pos
end

local function process_mod_icon(mod)
	local atlas, pos = get_mod_icon_info(mod)
	local key = "icon_" .. (mod.id or mod.prefix or mod.name or 'mod')
	local card = {
		key = key,
		set = atlas,
		atlas = atlas,
		pos = pos,
	}
	output_image(card)
	return "images/" .. key:gsub("?", "_") .. ".png"
end

return process_mod_icon

File: mod/library/server/handle-message/exporter/process-mod/process-mod-icon.ts
import outputImage from "../_common/output-image.ts";

/**
 *
 * @param mod
 */
const processModIcon = (mod: any): string => {
	let atlas = "modicon";

	if (mod.prefix !== undefined && G.ASSET_ATLAS[`${mod.prefix}_modicon`] !== undefined) {
		atlas = `${mod.prefix}_modicon`;
	}

	let pos = {
		x: 0,
		y: 0
	};

	if (!mod.can_load && mod.load_issues && mod.load_issues.prefix_conflict) {
		atlas = "mod_tags";
		pos = {
			x: 0,
			y: 0
		};
	}

	const key = `icon_${mod.id || mod.prefix || mod.name || "mod"}`;

	const card = {
		atlas,
		key,
		pos,
		set: atlas
	};

	outputImage(card);

	return `${key.replace("?", "_")}.png`;
};

export default processModIcon;
